File: 0543-diameter-of-binary-tree/0543-diameter-of-binary-tree.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */


int dia =0;
int deep(struct TreeNode* node){
    if(node == NULL) return 0;
    int right = deep(node->right);
    int left = deep(node->left);
 if (left + right > dia) {
        dia = left + right;
    }

    return (left > right ? left : right) + 1;
}
int diameterOfBinaryTree(struct TreeNode* root) {
    dia =0;
    deep(root);
    return dia;
}

File: 0543-diameter-of-binary-tree/0543-diameter-of-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int diameter = 0;
    public int diameterOfBinaryTree(TreeNode root) {
        depthtree(root);   
        return diameter;     
    }
    public int depthtree(TreeNode node){
        if(node==null) return 0;

        int left = depthtree(node.left);
        int right = depthtree(node.right);

        diameter=(Math.max(diameter,left+right));

        return Math.max(left,right)+1;

    }
}